content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,188 @@
+
+#### 3. **I/O Wait (avg_io_wait)**
+
+- **Node-1** experienced minimal I/O wait, from **0.0002 to 0.0077 seconds**, while **Node-2** showed no I/O wait when handling lower loads.
+- The **Single Node configuration** maintained **0.0002 seconds** for I/O wait, reflecting efficient I/O handling even under increased loads.
+
+#### 4. **CPU Wait (avg_cpu_wait)**
+
+- **Node-1** in the two-node setup had a CPU wait ranging from **1.045 to 1.9356 seconds** during heavier query processing, while **Node-2** had a much lower CPU wait from **1.0964 to 0.0039 seconds**.
+- In **Single Node mode**, CPU wait times increased dramatically, ranging from **1.8338 to 9.1992 seconds**, indicating contention for CPU resources under increased query load.
+
+#### 5. **Disk Reads (avg_read_from_disk)**
+
+- Disk reads for **Node-1** ranged from **13,768,499.2 to 21,592,132.27 bytes** in the two-node configuration, with **Node-2** reading **7,768,951.47 to 0 bytes** depending on load distribution.
+- In **Single Node configuration**, disk reads increased significantly, varying from **3,258,299.73 to 11,590,451.2 bytes** under solo operation, revealing increased reliance on disk when a single node handled queries.
+
+#### 6. **Filesystem Reads (avg_read_from_filesystem)**
+
+- **Filesystem reads on Node-1** ranged from **14,377,664.77 to 22,227,520.87 bytes** in the two-node setup, while **Node-2** saw reads between **8,374,316.53 and 589,784.78 bytes**.
+- In **Single Node configuration**, filesystem reads ranged from **32,649,785.97 to 30,984,717.98 bytes**, showing a notable increase compared to the two-node configuration.
+
+#### 7. **Memory Usage (avg_memory_tracked)**
+
+- **Memory tracked** for **Node-1** ranged from **727,494,761.07 to 956,479,931 bytes**, and **Node-2** from **819,671,565.67 to 725,970,944 bytes** in the two-node setup.
+- In **Single Node mode**, memory usage varied from **877,527,596.88 to 903,618,884.9 bytes**, indicating increased memory demand when handling queries without load balancing.
+
+#### 8. **Selected Rows per Second (avg_selected_rows_per_second)**
+
+- **Node-1** showed a row selection rate from **412,713.28 to 648,431.85 rows/sec**, while **Node-2** had rates from **232,168.82 to 238.1 rows/sec** in the two-node setup.
+- In **Single Node mode**, selected rows per second varied widely from **458,760 to 9,742,907.23 rows/sec**, demonstrating the node’s ability to manage high query demands under load.
+
+#### 9. **Inserted Rows per Second (avg_inserted_rows_per_second)**
+
+- The **inserted rows per second** remained low on **Node-1** with **246.08 to 2,006 rows/sec** and on **Node-2** with **253.93 to 238.1 rows/sec** in the two-node setup.
+- In **Single Node mode**, insertion rates were steady from **214.92 to 222 rows/sec**, suggesting minimal impact on insertions despite high query activity.
+
+<!-- markdownlint-enable MD024 -->
+
+### Incremental Comparison of Key Metrics Across Configurations
```

Comment:
None of these setups are production grade(1 or 2 nodes), so, I do not think these numbers are worth comparing. What cluster sizing did we have for ingestion and read-throughput benchmarks?

Also, it would help to understand the type and size of the EC2 machine to determine whether the instance itself limits these benchmarks.